generate-artifacts.ps1

```diff
@@ -0,0 +1,75 @@
+param(
+    [Parameter(Mandatory=$true, ValueFromPipeline=$true)]
+    [string]$ArtifactsStorageAccountName
+)
+
+# default script values
+$rgName = "mate-resources"
+$containerName = "task-artifacts"
+$resourcesTemplateName = "exported-template.json"
+$taskName = "task1"
+$tempFolderPath = "$PWD/temp"
+$artifactsConfigPath = "$PWD/artifacts.json"
+
+# initial validation
+Write-Output "Running initial validation"
+$context = Get-AzContext  
+if ($context)   
+{  
+    Write-Output "Azure Powershell module is installed, account is connected."  
+} else {  
+    throw "Please log in to Azure using Azure Powershell module (run Connect-AzAccount)"
+}  
+
+Write-Output "Checking if storage account exists"
+$storageAccount = Get-AzStorageAccount -Name $ArtifactsStorageAccountName -ResourceGroupName $rgName -ErrorAction SilentlyContinue
+if ($storageAccount) {
+    Write-Output "Storage account found"
+} else { 
+    throw "Unable to find storage account $ArtifactsStorageAccountName in the resource group $rgName. Please make sure, that you specified the correct name of the storage account"
+}
+
+Write-Output "Checking if artifacts storage container exists" 
+$artifactContainer = Get-AzStorageContainer -Name $containerName -Context $storageAccount.Context -ErrorAction SilentlyContinue
+if ($artifactContainer) { 
+    Write-Output "Storage container for artifacts found!" 
+} else { 
+    throw "Unable to find a storage container $containerName in the storage account $ArtifactsStorageAccountName, please make sure that it's created"
+}
+
+
+# generation of artifacts
+Write-Output "Generating artifacts"
+
+Write-Output "Checking if temp folder exists"
+if (-not (Test-Path "$tempFolderPath")) { 
+    Write-Output "Temp folder does not exist, creating..."
+    New-Item -ItemType Directory -Path $tempFolderPath
+}
+
+Write-Output "Exporting resources template"
+Export-AzResourceGroup -ResourceGroupName $rgName -Path "$tempFolderPath/$resourcesTemplateName" -Force
```

Comment:
Ensure that the `Export-AzResourceGroup` cmdlet is available and correctly used. This cmdlet might require additional parameters or a different approach depending on your Azure setup.